### ISSUE TITLE
Failed test_agent_groups System tests in Enrollment_Cluster on 4.8.0 Beta 2

### ISSUE THREAD

**Issue by Deblintrake09** [wazuh/wazuh-qa#5041]
## Description

During the testing in [Release 4.8.0 - Beta 2 - System tests](https://github.com/wazuh/wazuh/issues/22123), multiple test cases have failed in the `enrollment_cluster` environment, with tests located in the `test_agent_groups` suite.

The test was executed in a Ubuntu 22.04.2 LTS T3.Large 30GB SSD, provided by the devops team.

## Current behavior

The following test cases failed:
|Tests| 	Failed| 	Success| 
|--|--|--|
|test_cluster/test_agent_groups/test_agent_default_group_added.py 	|1 	|2 	|
|test_cluster/test_agent_groups/test_assign_groups_guess.py 	|4 	|8 	|
|test_cluster/test_agent_groups/test_group_hash.py 	|1 	|7 	|

**Evidence Report:** [enrollment_cluster.zip](https://github.com/wazuh/wazuh/files/14409525/enrollment_cluster.zip)

## Expected results

All the tests should be passed

More research is needed

**Comment by rauldpm** [comment]
```
E           AssertionError:  Did not recieve expected agent status 002  Agent-1708961985  172.18.0.5  active in data ID   NAME              IP          STATUS           VERSION       NODE NAME     
E           000  wazuh-master      127.0.0.1   active           Wazuh v4.8.0  wazuh-master  
E           002  Agent-1708961985  172.18.0.5  never_connected  unknown       unknown       
E           003  wazuh-agent1      172.18.0.5  active           Wazuh v4.8.0  wazuh-master
```

```
E                   TimeoutError: Did not found the expected callback in wazuh-master: .*Group assigned: 'group_test'
```

```
E                   TimeoutError: Did not found the expected callback in wazuh-master: .*Group assigned: 'group_test'
```

```
E                   TimeoutError: Did not found the expected callback in wazuh-master: .*Group assigned: 'default,group_test'
```

```
E                   TimeoutError: Did not found the expected callback in wazuh-master: .*Group assigned: 'default,group_test'
```

```
E           AssertionError: wazuh-master reported different global group hash
E           assert 'c80958729942...8491686af2dc8' == 'fbfd54157e38...cb28af49420b8'
E             - fbfd54157e38cb06f36da1e3985cb28af49420b8
E             + c8095872994278a6c853d33dd1c8491686af2dc8
``

**Comment by juliamagan** [comment]
During the analysis it could be seen that the errors reported were due to the new environment, except for the `test_assign_groups_guess` error that we had to apply a fix in the internal options.

In addition, it was found that the `test_agent_groups` test said that it had passed correctly, but it had not really been launched. When it was launched, it was found that this test was failing, together with `test_assign_agent_to_a_group`. To solve the error of these tests, the timeout has been extended.
